lib/services/resource.go

```diff
@@ -335,11 +335,8 @@ func (m *Metadata) SetTTL(clock clockwork.Clock, ttl time.Duration) {
 	m.Expires = &expireTime
 }
 
-// CheckAndSetDefaults checks validity of all parameters and sets defaults
-func (m *Metadata) CheckAndSetDefaults() error {
-	if m.Name == "" {
-		return trace.BadParameter("missing parameter Name")
-	}
+// SetDefaults sets default values if missing
+func (m *Metadata) SetDefaults() {
```

Comment:
this is a common method name, so can you bring it back as well as it's return value

Reply:
ah, misread that

Reply:
actually, let's not complicate things - let's have only one method `CheckAndSetDefaults` that does both things for now.

Reply:
This is a `new` method used to SetDefaults and returns no errors as opposed to CheckAndSetDefaults. This method is then used in a few other places, like `Equals`

Reply:
It's also used in V2 -> V3 converter `func (r *RoleV2) V3() *RoleV3 `